core/impl/src/main/java/com/blazebit/persistence/impl/function/datetime/epoch/EpochFunction.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datetime.epoch;
+
+import com.blazebit.persistence.spi.FunctionRenderContext;
+import com.blazebit.persistence.spi.JpqlFunction;
+import com.blazebit.persistence.spi.TemplateRenderer;
+
+/**
+ * @author Moritz Becker (moritz.becker@gmx.at)
+ * @since 1.2
+ */
+public abstract class EpochFunction implements JpqlFunction {
+
+    protected final TemplateRenderer renderer;
+
+    public EpochFunction(String template) {
+        this.renderer = new TemplateRenderer(template);
+    }
+
+    @Override
+    public boolean hasArguments() {
+        return true;
+    }
+
+    @Override
+    public boolean hasParenthesesIfNoArguments() {
+        return true;
+    }
+
+    @Override
+    public Class<?> getReturnType(Class<?> firstArgumentType) {
+        return Integer.class;
```

Comment:
The type for epoch is commonly integer - though this is also the reason for the year 2038 problem.